snapcraft/parts/lifecycle.py

```diff
@@ -160,11 +160,13 @@ def _run_command(  # noqa PLR0913 # pylint: disable=too-many-branches, too-many-
 
     track_stage_packages = getattr(parsed_args, "enable_manifest", False)
 
+    # notice the "base" snap does not have a base, so we need to use the
+    # build-base instead, which is allowed to be "devel"
     lifecycle = PartsLifecycle(
         project.parts,
         work_dir=work_dir,
         assets_dir=assets_dir,
-        base=project.get_effective_base(),
+        base=project.get_effective_base(allow_return_devel=project.type == "base"),
```

Comment:
I would remove this signature change and deal with the variance from the type in get_effective_base inside the method as the object is itself project and can access its type

Reply:
I agree, it's best to hide this logic inside `Project.get_effective_base()` and `utils.get_effective_base()`

Reply:
I thought about it at first, but I don't know why it was split into 2 parts in different places. Will move it to utils